File: apps/ios/LanjingQuiz/Models/BankMeta.swift
import Foundation

/// Persisted in the bank's meta.json. counts (per-category) match the JSONL
/// line counts exactly and are the category-list data source. `targets` is
/// informational only — the web collector once wrote a typo there
/// ("语言理解" vs the real file 言语理解.jsonl), so the client never derives
/// file names from it. `papers` records the crawled paper ids (value true) so
/// an interrupted crawl resumes without re-entering completed papers.
struct BankMeta: Codable, Equatable, Sendable {
    let version: Int?
    let round: Int?
    let lastRun: String?
    let targets: [String]?
    let counts: [String: Int]?
    let papers: [String: Bool]?

    init(
        version: Int? = nil,
        round: Int? = nil,
        lastRun: String? = nil,
        targets: [String]? = nil,
        counts: [String: Int]? = nil,
        papers: [String: Bool]? = nil
    ) {
        self.version = version
        self.round = round
        self.lastRun = lastRun
        self.targets = targets
        self.counts = counts
        self.papers = papers
    }

    var totalCount: Int { counts?.values.reduce(0, +) ?? 0 }
}

File: apps/android/LanjingQuiz/src/main/java/com/qzh/lanjingquiz/Data/BankMeta.kt
package com.qzh.lanjingquiz.Data

import kotlinx.serialization.Serializable

/**
 * 题库元信息(bank/meta.json;spec §3.2 键逐字:version/round/lastRun/targets/counts/papers)。
 * counts 与 JSONL 行数一致,是分类列表数据源;targets 仅信息性 —— 文件名字恒从硬编码
 * BankLogic.categories 派生,绝不从 targets 派生(历史上收集器在 targets 写过错别字 "语言理解")。
 * papers 记录已爬卷 paperId → true(断点续爬)。
 */
@Serializable
data class BankMeta(
    val version: Int = 1,
    val round: Int = 0,
    val lastRun: String? = null,
    val targets: List<String> = emptyList(),
    val counts: Map<String, Int> = emptyMap(),
    val papers: Map<String, Boolean> = emptyMap(),
) {
    val totalCount: Int get() = counts.values.sum()
}
